### ISSUE TITLE
[MRRESOURCES-37] embedded error with maven-remote-resources-plugin; maven build order?

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-remote-resources-plugin#160]
**[jieryn](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jieryn)** opened **[MRRESOURCES-37](https://issues.apache.org/jira/browse/MRRESOURCES-37?redirect=false)** and commented

I am attaching a sample project which exhibits the error. Please run   mvn -f pom-firstTime.xml  to prime your .m2/repository so that the real pom.xml's \<build> finds the artifacts properly. Now, run mvn (default goal install is coded) and notice the Embedded error. I scan the previously created JARs in my .m2/repository and the maven-remote-resources-plugin files are definitely in the JAR.....

I am at a loss, not convinced this is a bug, but need better minds to examine this. Thank you!

---

**Affects:** 1.0

**Attachments:**
- [MNG-4094.zip](https://issues.apache.org/jira/secure/attachment/12718310/MNG-4094.zip) (_45.66 kB_)
- [testing.zip](https://issues.apache.org/jira/secure/attachment/12718315/testing.zip) (_21.67 kB_)

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

here is a structure that works around it

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

When adding the following to the pluginManagement sections in both POMs, I cannot reproduce the problem:

```xml
<plugin>
  <artifactId>maven-remote-resources-plugin</artifactId>
  <version>1.1-SNAPSHOT</version>
</plugin>
```

Actually, even if I add this section below to the top-level POMs (as above), I still cannot reproduce the problem:

```xml
<plugin>
  <artifactId>maven-remote-resources-plugin</artifactId>
  <version>1.0</version>
</plugin>
```

If I remove the directory completely and remove all com/acme/** from the local repository, then unarchive the testing.zip file again, only then can I reproduce the problem.

I'm not sure what to do with this, but it seems plain to me that this is working if you specify 1.1-SNAPSHOT for the version of the remote-resources plugin (the version we're working on now).

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

interesting comment here:

http://www.mail-archive.com/jboss-dev@list.working-dogs.com/msg06762.html

This fits what's going on in this test project...the shared-resource zip archive is being replaced during the build. Notice that only in one of the later projects is the exception being thrown, which I think means it's happening after the archive is replaced under the existing classloader...

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

I've verified that the file in question is changing during the build, which is throwing off the cached URLConnection data (apparently this is what's happening; it matches the link I posted to a tee).

I'm not aware of any way we could compensate for something like this happening down deep in the com.sun.** classes, and I've tried a few things, without any luck. My advice would be to adopt something closer to the project layout Brett described above.
